library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -408,10 +408,12 @@ private Uri insertDownload(Uri uri, ContentValues values, SQLiteDatabase db, int
                 dest = DownloadsDestination.DESTINATION_CACHE_PARTITION;
             }
             if (dest == DownloadsDestination.DESTINATION_FILE_URI) {
-                getContext().enforcePermission(
-                        android.Manifest.permission.WRITE_EXTERNAL_STORAGE, Binder.getCallingPid(), Binder.getCallingUid(),
-                        "need WRITE_EXTERNAL_STORAGE permission to use DESTINATION_FILE_URI");
-                checkFileUriDestination(values);
+                if (!getFileUriDestinationPath(values).contains("/" + getContext().getPackageName() + "/")) {
```

Comment:
Agh, you're right (Assume you mean external). It's something like that and will contain the package.

Our use case only wants to use internal storage and we don't want to request the external permission. I'll try to find a reliable and correct (!) way of determining which kind of storage is being used within this method. I'll close this PR - thanks (and sorry!) for you time.